### PR TITLE
Added Interoperability information of EFGS

### PR DESCRIPTION
**Pull request by emmetsap**

### CODE REVIEW

**Comment by MikeMcC399** [comment]
Please do a spell and grammar check on the additions to solution_architecture.md

seperate relivance are spelled incorrectly (separate, relevance) and 

countries backend should be singular and possessive (country's backend) because it is each country's backend, not a combined backend of all countries together.

**Comment by emmetsap** [comment]
Updated to correct grammar

File: solution_architecture.md [hunk 1]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+Most European countries are developing similar contact tracing apps, these app may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications.
```
Comment (ebratter) [inline solution_architecture.md:274]: Suggested rewording:
Most European countries are developing similar contact tracing apps. These apps transmit and detect GAEN format diagnosis keys through the use of the common frameworks developed by Google and Apple.

File: solution_architecture.md [hunk 2]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+Most European countries are developing similar contact tracing apps, these app may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications. 
+Each country has its own separate database of users, in order to coordinate exposure information between countries a common service is required.
+The European Federation Gateway Service (EFGS) handles sharing diagnosis keys between different country's backend servers.
```
Comment (ebratter) [inline solution_architecture.md:276]: Suggested rewording:
The European Federation Gateway Service (EFGS) enables interoperability of diagnosis keys between the connected country backend servers.

File: solution_architecture.md [hunk 3]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+
+![Figure 15: High-level EFGS overview](images/solution_architecture/EFGS_overview.jpg "Figure 15: High-level EFGS overview")
+
+The Federation Gateway Service facilitates backend-to-backend integration, countries can onboard incrementally, while the national backends retain flexibility and control over data distribution to their users.
+For example if a German user is visiting France then all uploaded keys are of relevance to the French database, likewise if a French user is visiting Germany that users keys are of relevance to the German database.
```
Comment (ebratter) [inline solution_architecture.md:281]: Suggested rewording:
For example, if a German citizen visits France and then becomes infected, the keys of the German citizen are then relevant for the citizens of France. In this case the German citizen keys would be shared with the EFGS to enable the French backend to obtain the keys. Similarly, if a French user is visiting Germany that users keys are of relevance to the German database.

File: solution_architecture.md [hunk 4]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+
+![Figure 16: Autonomous National Backend](images/solution_architecture/EFGS_Autonomous_Backend.jpg "Figure 16: Autonomous National Backend")
+
+In the example above user A from country A travels to country B and tests positive, only the relevant users in Country B will receive the alert.
+Devices only communicate with their country's backend, the EFGS is only sent keys by a country's backend.
```
Comment (ebratter) [inline solution_architecture.md:286]: Suggested rewording;
Devices only communicate with their country backend. The country backend is then responsible to send relevant keys to the EFGS.

File: solution_architecture.md [hunk 5]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+All countries provide all keys to the EFGS, the EFGS only transmits relevant keys to each country's backend, notifications and alerts are handled by each individual country's backend.
```
Comment (ebratter) [inline solution_architecture.md:287]: Suggested rewording:
All connected countries provide keys to the EFGS. The EFGS then makes available relevant keys to each additional connected country backend. Notifications and alerts are handled by each individual country backend.

File: solution_architecture.md [hunk 6]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+Most European countries are developing similar contact tracing apps, these app may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications. 
+Each country has its own separate database of users, in order to coordinate exposure information between countries a common service is required.
```
Comment (ebratter) [inline solution_architecture.md:275]: We do not have a database of users we should be careful about the wording here. Suggested rewording:
Each country has its own separate database which contain the keys from infected individuals. In order to coordinate exposure information between countries a common service is required to enable interoperability.

File: solution_architecture.md [hunk 7]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+The European Federation Gateway Service (EFGS) handles sharing diagnosis keys between different country's backend servers.
+
+![Figure 15: High-level EFGS overview](images/solution_architecture/EFGS_overview.jpg "Figure 15: High-level EFGS overview")
+
+The Federation Gateway Service facilitates backend-to-backend integration, countries can onboard incrementally, while the national backends retain flexibility and control over data distribution to their users.
```
Comment (ebratter) [inline solution_architecture.md:280]: Might also consider linking to the EFGS gitrepo?

File: solution_architecture.md [hunk 8]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+For example if a German user is visiting France then all uploaded keys are of relevance to the French database, likewise if a French user is visiting Germany that users keys are of relevance to the German database.
+
+![Figure 16: Autonomous National Backend](images/solution_architecture/EFGS_Autonomous_Backend.jpg "Figure 16: Autonomous National Backend")
+
+In the example above user A from country A travels to country B and tests positive, only the relevant users in Country B will receive the alert.
```
Comment (ebratter) [inline solution_architecture.md:285]: Suggested rewording:
In the example above, user A from country A travels to country B and tests positive, only the relevant users (those which came within proximity of the infected user A) in Country B will receive the alert.

**Review by ebratter — CHANGES_REQUESTED** [review]

File: solution_architecture.md (hunk 7 above)
Comment (emmetsap) [reply solution_architecture.md:280]: Added link to text "European Federation Gateway Service (EFGS)"

File: solution_architecture.md (hunk 1 above)
Comment (emmetsap) [reply solution_architecture.md:274]: This change makes the assumption all EU countries will follow the standard GAEN framework, I presume they will but didn't want to declare it a certainty.

Comment (MikeMcC399) [reply solution_architecture.md:274]: The list [Mobile contact tracing apps in EU Member States](https://ec.europa.eu/info/live-work-travel-eu/health/coronavirus-response/travel-during-coronavirus-pandemic/mobile-contact-tracing-apps-eu-member-states_en) from the EU Commission shows which countries are using the GAEN framework in the column labelled "Interoperable - is this app potentially interoperable?"

Most countries list "Yes", only France and Hungary currently list "No" and there are five countries which have no app.

You could change "are developing" to "have developed" and you might like to add the link to the EU Commission's page into the document.

Comment (MikeMcC399) [inline solution_architecture.md:274]: these app => these apps (plural)

File: solution_architecture.md [hunk 9]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+Each country has its own separate database which contain the keys from infected individuals. In order to coordinate exposure information between countries a common service is required to enable interoperability.
+The [European Federation Gateway Service (EFGS)](https://github.com/eu-federation-gateway-service/efgs-federation-gateway) enables interoperability of diagnosis keys between the connected country backend servers.
+
+![Figure 15: High-level EFGS overview](images/solution_architecture/EFGS_overview.jpg "Figure 15: High-level EFGS overview")
+
+The Federation Gateway Service facilitates backend-to-backend integration, countries can onboard incrementally, while the national backends retain flexibility and control over data distribution to their users.
+For example, if a German citizen visits France and then becomes infected, the keys of the German citizen are then relevant for the citizens of France. In this case the German citizen keys would be shared with the EFGS to enable the French backend to obtain the keys. Similarly, if a French user is visiting Germany that users keys are of relevance to the German database.
```
Comment (MikeMcC399) [inline solution_architecture.md:281]: that users keys are of relevance => that user's keys are of relevance (possessive)

File: solution_architecture.md [hunk 10]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+Most European countries are developing similar contact tracing apps, these app may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications. 
+Each country has its own separate database which contain the keys from infected individuals. In order to coordinate exposure information between countries a common service is required to enable interoperability.
```
Comment (roesslerj) [inline solution_architecture.md:275]: which contain => which contains
between countries a common => between countries, a common

File: solution_architecture.md [hunk 11]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+The [European Federation Gateway Service (EFGS)](https://github.com/eu-federation-gateway-service/efgs-federation-gateway) enables interoperability of diagnosis keys between the connected country backend servers.
+
+![Figure 15: High-level EFGS overview](images/solution_architecture/EFGS_overview.jpg "Figure 15: High-level EFGS overview")
+
+The Federation Gateway Service facilitates backend-to-backend integration, countries can onboard incrementally, while the national backends retain flexibility and control over data distribution to their users.
```
Comment (roesslerj) [inline solution_architecture.md:280]: integration, countries => integration. Countries

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: solution_architecture.md (hunk 1 above)
Comment (emmetsap) [reply solution_architecture.md]: I have included the link you suggested, I would prefer to keep "are developing" as there are development changes ongoing with all apps and on the iOS Android side.

File: solution_architecture.md [hunk 12]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+[Most European countries are developing similar contact tracing apps](https://ec.europa.eu/info/live-work-travel-eu/health/coronavirus-response/travel-during-coronavirus-pandemic/mobile-contact-tracing-apps-eu-member-states_en), these apps may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications.
```
Comment (roesslerj) [inline solution_architecture.md:274]:
```suggestion
[Most European countries are developing similar contact tracing apps](https://ec.europa.eu/info/live-work-travel-eu/health/coronavirus-response/travel-during-coronavirus-pandemic/mobile-contact-tracing-apps-eu-member-states_en). These apps may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications. 
```

File: solution_architecture.md [hunk 13]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
 
 ## Cross-Border Interoperability
 
-A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the framework by Apple and Google is being used.
+A definite prerequisite for compatibility is that the identifiers of the mobile devices can be matched, i.e. the GAEN framework by Apple and Google is being used.
 
-Further details will be added as soon as they are available.
+[Most European countries are developing similar contact tracing apps](https://ec.europa.eu/info/live-work-travel-eu/health/coronavirus-response/travel-during-coronavirus-pandemic/mobile-contact-tracing-apps-eu-member-states_en), these apps may use the common frameworks by Google and Apple, enabling transmission and detection of GAEN format diagnosis keys between devices running different contact tracing applications. 
+Each country has its own separate database which contains the keys from infected individuals. In order to coordinate exposure information between countries, a common service is required to enable interoperability.
```
Comment (roesslerj) [inline solution_architecture.md:275]:
```suggestion
Each country has its own separate database, which contains the keys from infected individuals. In order to coordinate exposure information between countries, a common service is required to enable interoperability.
```

File: solution_architecture.md [hunk 14]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+The [European Federation Gateway Service (EFGS)](https://github.com/eu-federation-gateway-service/efgs-federation-gateway) enables interoperability of diagnosis keys between the connected country backend servers.
+
+![Figure 15: High-level EFGS overview](images/solution_architecture/EFGS_overview.jpg "Figure 15: High-level EFGS overview")
+
+The Federation Gateway Service facilitates backend-to-backend integration. Countries can onboard incrementally, while the national backends retain flexibility and control over data distribution to their users.
+For example, if a German citizen visits France and then becomes infected, the keys of the German citizen are then relevant for the citizens of France. In this case the German citizen keys would be shared with the EFGS to enable the French backend to obtain the keys. Similarly, if a French user is visiting Germany that user's keys are of relevance to the German database.
```
Comment (roesslerj) [inline solution_architecture.md:281]:
```suggestion
For example, if a German citizen visits France and then becomes infected, the keys of the German citizen are then relevant for the citizens of France. In this case the German citizen keys would be shared with the EFGS to enable the French backend to obtain the keys. Similarly, if a French user is visiting Germany, that user's keys are of relevance to the German database.
```

File: solution_architecture.md [hunk 15]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+
+![Figure 16: Autonomous National Backend](images/solution_architecture/EFGS_Autonomous_Backend.jpg "Figure 16: Autonomous National Backend")
+
+In the example above, user A from country A travels to country B and tests positive, only the relevant users (those which came within proximity of the infected user A) in Country B will receive the alert.
```
Comment (roesslerj) [inline solution_architecture.md:285]:
```suggestion
In the example above, user A from country A travels to country B and afterwards tests positive. Only the relevant users (those which came within proximity of the infected user A) in Country B will receive the alert.
```

File: solution_architecture.md [hunk 16]
```diff
@@ -269,9 +269,24 @@ If the back end calls from the mobile applications cannot be spread as evenly as
...
+Devices only communicate with their country's backend. That country's backend is then responsible to send relevant keys to the EFGS.
+All connected countries provide keys to the EFGS. The EFGS then makes available relevant keys to each additional connected country's backend. Notifications and alerts are handled by each individual country's backend.
+The EFGS stores information of all currently infected citizens along with a list of countries they visited.
+In order for the EFGS to function correctly all users must specify their visited countries correctly (either manually or automatically).
```
Comment (roesslerj) [inline solution_architecture.md:289]:
```suggestion
In order for the EFGS to function correctly, all users must specify their visited countries correctly (either manually or automatically). 
```

**Review by roesslerj — CHANGES_REQUESTED** [review]
Can we add the sources of the JPEGs (in whatever program you created them), so that future changes do not need to recreate them again? Also in EFGS_overview: 
" ... _and_ data conversion" at the bottom.

**Review by roesslerj — APPROVED** [review]

**Pull request merged**